bftengine/src/bftengine/messages/PrePrepareMsg.cpp

```diff
@@ -194,6 +196,29 @@ bool PrePrepareMsg::checkRequests() const {
   Assert(false);
   return false;
 }
+std::string PrePrepareMsg::getMessageClientCorrelationId(int index) {
+  auto it = RequestsIterator(this);
+  int req_num = 0;
+  while (!it.end() && req_num < index) {
+    it.gotoNext();
+    req_num++;
+  }
+  char* requestBody = nullptr;
+  it.getCurrent(requestBody);
+  ClientRequestMsg req((ClientRequestMsgHeader*)requestBody);
+  return req.getCid();
+}
+std::string PrePrepareMsg::getAllRequestsCorrelationIdAsString() {
+  std::stringstream ret;
+  ;
+  auto it = RequestsIterator(this);
+  char* requestBody = nullptr;
+  while (it.getAndGoToNext(requestBody)) {
+    ClientRequestMsg req((ClientRequestMsgHeader*)requestBody);
+    ret << req.getCid() << ";";
```

Comment:
[Minor]: Would having a trailing `;` in the output be a problem?

Reply:
I don't think so and I would prefer to leave it simple as we can...

Reply:
Great, makes sense.